src/charm.py

```diff
@@ -144,45 +162,30 @@ def _update_fiveg_n3_relation_data(self) -> None:
             )
 
     def _network_attachment_definitions_from_config(self) -> list[NetworkAttachmentDefinition]:
-        """Returns list of Multus NetworkAttachmentDefinitions to be created based on config."""
-        access_nad_config = {
-            "cniVersion": "0.3.1",
-            "ipam": {
-                "type": "static",
-                "routes": [
-                    {
-                        "dst": self._get_gnb_subnet_config(),
-                        "gw": self._get_access_network_gateway_ip_config(),
-                    },
-                ],
-                "addresses": [
-                    {
-                        "address": self._get_access_network_ip_config(),
-                    }
-                ],
-            },
-            "capabilities": {"mac": True},
-        }
-        if (access_interface := self._get_access_interface_config()) is not None:
+        """Returns list of Multus NetworkAttachmentDefinitions to be created based on config.
+
+        Checks if the config settings are correct before creating any NAD.
+        If the NAD config is invalid, returns None as a protection not to process invalid configs.
+        """
+        if invalid_configs := self._get_invalid_configs():
+            self.unit.status = BlockedStatus(
+                f"The following configurations are not valid: {invalid_configs}"
+            )
+            return None  # type: ignore
```

Comment:
This should not return `None`, but an empty list according to the type hint. 

I understand however that this might be necessary in order to not remove all `NADs` if the config is updated to an invalid state. In that case, the type hints should be updated here and in the lib.

Reply:
I will update the type hints.

Reply:
Done